tests/cephfs/snapshot_clone/snap_schedule_with_mds_restart.py

```diff
@@ -0,0 +1,259 @@
+        ]
+
+        for cmd in modified_commands:
+            client1.exec_command(sudo=True, cmd=cmd)
+        sleep(300)
+        verify_snap_schedule(
+            client1,
+            f"{fuse_mounting_dir_1}dir_fuse/",
+            fs_name=default_fs,
+            schedule=f"1{m_granularity}",
+        )
+        verify_snap_schedule(
+            client1,
+            f"{kernel_mounting_dir_1}dir_kernel/",
+            fs_name=default_fs,
+            schedule=f"1{m_granularity}",
+        )
+        for i in range(1, 5):
+            client1.exec_command(
+                sudo=True, cmd=f"umount {fuse_mounting_dir_1} {kernel_mounting_dir_1}"
+            )
+            fs_util.fuse_mount(
+                [client1],
+                fuse_mounting_dir_1,
+                extra_params=f"--client_fs {default_fs}",
+            )
+            retry_mount(
+                [client1],
+                kernel_mounting_dir_1,
+                ",".join(mon_node_ips),
+                extra_params=f",fs={default_fs}",
+            )
+            client1.exec_command(
+                sudo=True, cmd=f"stat {fuse_mounting_dir_1}dir_fuse/.snap/"
+            )
+            client1.exec_command(
+                sudo=True, cmd=f"stat {kernel_mounting_dir_1}dir_kernel/.snap/"
+            )
+
+            fs_util.get_active_mdss(client1, default_fs)
+            mds = fs_util.get_active_mdss(client1, default_fs)
+            mds_active_node = ceph_cluster.get_node_by_hostname(mds[0].split(".")[1])
+            log.info("Get the mdthresh_evicted ceph mds perf cntr value")
+            fs_util.reboot_node_v1(mds_active_node)
+            fs_util.check_active_mds_count(client1, default_fs, 1)
+
+            client1.exec_command(
+                sudo=True, cmd=f"ls {fuse_mounting_dir_1}dir_fuse/.snap/"
+            )
+            client1.exec_command(
+                sudo=True, cmd=f"ls {kernel_mounting_dir_1}dir_kernel/.snap/"
+            )
+
+        return 0
+    except Exception as e:
+        log.error(e)
+        log.error(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Clean Up in progess")
+        commands = [
+            f"ceph fs snap-schedule deactivate path --fs {default_fs}",
+            f"ceph fs snap-schedule remove path --fs {default_fs}",
+            "ceph config set mgr mgr/snap_schedule/allow_m_granularity false",
+        ]
+        modified_commands = [
+            cmd.replace("path", item) for item in sanp_schedule_list for cmd in commands
+        ]
+        for cmd in modified_commands:
+            client1.exec_command(sudo=True, cmd=cmd)
+        commands = [
+            "ceph config set mon mon_allow_pool_delete true",
+            f"ceph fs volume rm {default_fs} --yes-i-really-mean-it",
+        ]
+        for cmd in commands:
+            client1.exec_command(sudo=True, cmd=cmd)
+
+
+def verify_snap_schedule(client, path, fs_name, schedule):
```

Comment:
Amar, all snap related functions are already present in - https://github.com/red-hat-storage/cephci/blob/master/tests/cephfs/snapshot_clone/cephfs_snap_utils.py, and the one you have created is also present in --  https://github.com/red-hat-storage/cephci/blob/master/tests/cephfs/snapshot_clone/cephfs_snap_utils.py#L263 - Can we reuse them instead of creating a new one ?

Reply:
+1

Reply:
Hi Hemanth,
Both had different functionality. 
verify_snap_schedule --> is checking if the `ceph fs snap-schedule list` list this added path
validate_snap_schedule --> Is verifying snapshots are getting created or not
So I have included both verifications